playback/src/decoder/symphonia_decoder.rs

```diff
@@ -131,6 +187,71 @@ impl SymphoniaDecoder {
         }
     }
 
+    pub(crate) fn local_file_metadata(&mut self) -> Option<LocalFileMetadata> {
+        let metadata = self.metadata()?;
+        let tags = metadata.current()?.tags();
+        let mut metadata = LocalFileMetadata::default();
+
+        for tag in tags {
+            if let Value::String(value) = &tag.value {
+                match tag.std_key {
+                    // We could possibly use mem::take here to avoid cloning, but that risks leaving
+                    // the audio item metadata in a bad state.
+                    Some(StandardTagKey::TrackTitle) => metadata.name = value.clone(),
+                    Some(StandardTagKey::Language) => metadata.language = value.clone(),
+                    Some(StandardTagKey::Artist) => metadata.artists = value.clone(),
+                    Some(StandardTagKey::AlbumArtist) => metadata.album_artists = value.clone(),
+                    Some(StandardTagKey::Album) => metadata.album = value.clone(),
+                    Some(StandardTagKey::TrackNumber) => {
+                        metadata.number = value.parse::<u32>().unwrap_or_default()
```

Comment:
That'll set "0" when an empty string or something malformed. Make it `None` instead?

Reply:
It's the same problem as trying to parse the artists into a `Vec<String>` instead of just a `String`, given an arbitrary file you really have no idea what could be in the tags... it could be `33`, `33/37`, `33 / 37`, `33 of 37`, `33 out of 37` and so on. I suppose in this example each case would work if you did what C++ does and just keep parsing numbers until you fail and then return a success if you parsed any characters.

It would be interesting to be able to get some statistics of files where it isn't just a number to see if it's worth it, though.